File: MyCypher.py
#!/usr/bin/env python2
#encoding: UTF-8

# Python Class for AES encryption
"""
    Example Usage
    enc_str = cipher.encrypt('secret')
    enc_str = cipher.decrypt(enc_str)
    print enc_str; #secret
"""

from Crypto.Cipher import AES
import base64
import hashlib
import sys


class MyCypher:
    # Default Key for encryption
    rawkey = 'asdfa923aksadsYahoasdw998sdsads'
    method = AES.MODE_CFB
    blocksize = 32  # 16, 32..etc
    padwith = '`'  # padding value for string

    # lambda function for padding
    def pad(self, s): return s + (self.blocksize - len(s) %
                                  self.blocksize) * self.padwith

    def __init__(self, iv, key=''):
        """
        construct for cypher class - get, set key and iv
        """
        if(not key):
            key = self.rawkey

        self.key = key
        self.iv = iv

    def getKEY(self):
        """
        get hashed key - if key is not set on init, then default key wil be used
        """
        if(not self.key):
            sys.exit()

        return hashlib.sha256(self.key.encode('utf-8')).hexdigest()[:32]

    def getIV(self):
        """
        get hashed IV value - if no IV values then it throw error
        """
        if(not self.iv):
            sys.exit()

        return hashlib.sha256(self.iv.encode('utf-8')).hexdigest()[:16]

    def encrypt(self, raw):
        """
        Encrypt given string using AES encryption standard
        """
        cipher = AES.new(self.getKEY(), self.method,
                         self.getIV(), segment_size=128)
        return base64.b64encode(cipher.encrypt(self.pad(raw)))

    """
    Decrypt given string using AES standard
    """

    def decrypt(self, encrypted):
        encrypted = base64.b64decode(encrypted)
        cipher = AES.new(self.getKEY(), self.method,
                         self.getIV(), segment_size=128)
        return cipher.decrypt(encrypted).decode('utf-8').rstrip(self.padwith)

File: MyCypher.php
<?php

/**
 * Valid encryption methods AES-256-CFB 
 * 
 * $cypher = new MyCypher($iv);
 * $php_encrypted      = $cypher->encrypt('test');
 * $php_decrypted      = $cypher->decrypt($php_encrypted);
 */
class MyCypher {

    private $key = 'asdfa923aksadsYahoasdw998sdsads';
    private $iv = null;
    private $method = "AES-256-CFB";
    private $blocksize = 32;
    private $padwith = '`';

    /*
     * construct for cypher class - get, set key and iv
     */

    function __construct($iv, $key = null) {

        if (is_string($key)) {
            $this->key = $key;
        }

        $this->iv = $iv;
    }

    /*
     * get hased key - if key is not set on init, then default key wil be used
     */

    private function getKEY() {

        if (empty($this->key)) {
            die('Key not set!');
        }

        return substr(hash('sha256', $this->key), 0, 32);
    }

    /*
     * get hashed IV value - if no IV values then it throw error
     */

    private function getIV() {

        if (empty($this->iv)) {
            die('IV not set!');
        }

        return substr(hash('sha256', $this->iv), 0, 16);
    }

    /*
     * Encrypt given string using AES encryption standard
     */

    public function encrypt($secret) {

        try {

            $padded_secret = $secret . str_repeat($this->padwith, ($this->blocksize - strlen($secret) % $this->blocksize));
            $encrypted_string = openssl_encrypt($padded_secret, $this->method, $this->getKEY(), OPENSSL_RAW_DATA, $this->getIV());
            $encrypted_secret = base64_encode($encrypted_string);
            return $encrypted_secret;
        } catch (Exception $e) {
            die('Error : ' . $e->getMessage());
        }
    }

    /*
     * Decrypt given string using AES standard
     */

    public function decrypt($secret) {
        try {
            $decoded_secret = base64_decode($secret);
            $decrypted_secret = openssl_decrypt($decoded_secret, $this->method, $this->getKEY(), OPENSSL_RAW_DATA, $this->getIV());
            return rtrim($decrypted_secret, $this->padwith);
        } catch (Exception $e) {
            die('Error : ' . $e->getMessage());
        }
    }

}
